src/Appwrite/Realtime/Realtime.php

```diff
@@ -0,0 +1,193 @@
+            if (isset($node['teamId']) && isset($node['roles'])) {
+                $roles[] = 'team:' . $node['teamId'];
+
+                foreach ($node['roles'] as $nodeRole) { // Set all team roles
+                    $roles[] = 'team:' . $node['teamId'] . '/' . $nodeRole;
+                }
+            }
+        }
+        return $roles;
+    }
+
+    /**
+     * Converts the channels from the Query Params into an array. 
+     * Also renames the account channel to account.USER_ID and removes all illegal account channel variations.
+     * 
+     * @param array $channels
+     */
+    static function parseChannels(array $channels)
+    {
+        $channels = array_flip($channels);
+
+        foreach ($channels as $key => $value) {
+            switch (true) {
+                case strpos($key, 'account.') === 0:
+                    unset($channels[$key]);
+                    break;
+                
+                case $key === 'account':
+                    if (!empty(self::$user->getId())) {
+                        $channels['account.' . self::$user->getId()] = $value;
+                    }
+                    unset($channels['account']);
+                    break;
+            }
+        }
+
+        if (\array_key_exists('account', $channels)) {
+            if (self::$user->getId()) {
+                $channels['account.' . self::$user->getId()] = $channels['account'];
+            }
+            unset($channels['account']);
+        }
+
+        return $channels;
+    }
+
+    /**
+     * Identifies the receivers of all subscriptions, based on the permissions and event.
+     * 
+     * @param array $event
+     * @param array $connections
+     * @param array $subscriptions
+     */
+    static function identifyReceivers(array &$event, array &$subscriptions)
+    {
+        $receivers = [];
+        if ($subscriptions[$event['project']]) {
+            foreach ($subscriptions[$event['project']] as $role => $subscription) {
+                foreach ($event['data']['channels'] as $channel) {
+                    if (
+                        \array_key_exists($channel, $subscriptions[$event['project']][$role])
+                        && (\in_array($role, $event['permissions']) || \in_array('*', $event['permissions']))
+                    ) {
+                        foreach (array_keys($subscriptions[$event['project']][$role][$channel]) as $ids) {
+                            $receivers[] = $ids;
+                        }
+                        break;
+                    }
+                }
+            }
+        }
+
+        return array_keys(array_flip($receivers));
+    }
+
+    /**
+     * Adds Subscription. 
+     * 
+     * @param string $projectId
+     * @param mixed $connection
+     * @param array $subscriptions
+     * @param array $roles
+     * @param array $channels
+     */
+    static function subscribe($projectId, $connection, $roles, &$subscriptions, &$connections, &$channels)
```

Comment:
Do we have any results for testing on small vs large amount of data?
Might be useful to list them here for future reference.

Reply:
Same for the unsubscribe.